docs/src/overview.md

```diff
@@ -42,7 +44,20 @@ ingredients. For example, a new `mesh` type can be created and implemented at
 first only for a specific solver. Thus, there is no need to consider all
 possible combinations of `mesh`es, `equations`, and `solver`s when implementing
 new features. This allows rapid prototyping of new ideas and is one of the main
-design goals behind Trixi.
+design goals behind Trixi. Below is a brief overview of the availability of
+different features on different mesh types.
+
+| Feature                                                      | [`TreeMesh`](@ref) | [`StructuredMesh`](@ref) | [`UnstructuredMesh2D`](@ref) | [`P4estMesh`](@ref) |
+|--------------------------------------------------------------|:------------------:|:------------------------:|:----------------------------:|:-------------------:|
+| Spatial dimension                                            |     1D, 2D, 3D     |        1D, 2D, 3D        |              2D              |        2D, 3D       |
+| Coordinates                                                  |      Cartesian     |        curvilinear       |          curvilinear         |     curvilinear     |
+| Connectivity                                                 |  *h*-nonconforming |        conforming        |          conforming          |  *h*-nonconforming  |
+| Element type                                                 |      hypercube     |         hypercube        |           hypercube          |      hypercube      |
```

Comment:
I don't know whether this is bikeshedding, but I do not like the name hypercube for the element type too much (it's ok for the domain). It is too mathematical a term and might confuse people into thinking - if not looking closely enough - that the meshes only support square/cube cells.

What about using `line/square/cube` for `TreeMesh` and `line, quadrilateral, hexahedron` (or `line, quad, hex` to not blow up the table) for the rest? As a side benefit, this would also make it more explicit that the unstructured solver only works on 2D meshes right now (in case someone misses the first row).
```suggestion
| Element type                                                 | line, square, cube |      line, quad, hex     |             quad             |   line, quad, hex   |
```

Reply:
I don't understand why the element types should be different for our current set of meshes. They all use the same reference element type?

Reply:
Agreed. Maybe it's just a matter of perspective: Coming from the engineering/CFD world, at least *to me* (not speaking for everyone), "element type" lets me think about "triangles vs. quads", not about "reference element type". So if this is intended to be a high-level overview as a first point of contact for new users, I wanted to put in the names that might trigger folks like me. I think the shape of the reference element is a detail that people in the know about how DG-type methods work will understand from that list anyways.

Reply:
If it's okay for you, I would like to avoid jargon and use `line, square, cube` for all 4 mesh types

Reply:
I prefer quad/hex to make the casual reader not think that this is a Cartesian mesh only - that's why I used the more general names to make these meshes stand out from the TreeMesh. But I'd be OK if you have a preference for square/cube.

Reply:
I guess it depends on the background you expect people to have. Someone with your engineering background will more likely recognize quad/hex. People coming from mathematics with a more theoretical background and just a solid foundation in numerical analysis, not necessarily experts on DG methods, will likely have no clue what's going on with all this witchcraft...

Reply:
Alright, let's keep it like this!

Reply:
To clarify: `line, quadrilateral, hexahedron` would be fine for all people as well, I guess - it's just a bit long...

Reply:
So what about quad*, hex* and an explanation *quad = quadrilateral, *hex = hexahedron below the table?